internal/expr/parser.go

```diff
@@ -392,47 +426,36 @@ func (p *Parser) parseColumn() (columnExpr, bool, error) {
 	return fullName{}, false, nil
 }
 
-func (p *Parser) parseFuncName() (columnExpr, bool, error) {
+func (p *Parser) parseFunc() (columnExpr, bool, error) {
 	cp := p.save()
-	if p.skipName() && p.skipByte('(') {
-		bracketLevel := 1
+	scp := p.startSubpart()
+	if p.skipName() && p.peekByte('(') {
 		for {
-			if p.pos == len(p.input) {
-				return funcExpr{}, false, nil
-			}
-
-			if ok, err := p.skipStringLiteral(); err != nil {
+			if err := p.advance(); err != nil {
+				cp.restore()
 				return funcExpr{}, false, err
-			} else if ok {
-				continue
 			}
-			if ok := p.skipComment(); ok {
-				continue
+
+			if p.bracketLevel == 0 {
+				break
 			}
 
-			if p.skipByte('(') {
-				// Keep track of bracket nesting level in function to identify
-				// closing bracket.
-				bracketLevel++
-				continue
+			if p.pos == len(p.input) {
+				break
 			}
-			if p.skipByte(')') {
-				bracketLevel--
-				if bracketLevel == 0 {
-					break
-				}
+
+			if in, ok, err := p.parseInputExpression(); err != nil {
+				cp.restore()
+				return funcExpr{}, false, err
+			} else if ok {
+				p.add(in)
 				continue
 			}
-			p.pos++
-		}
-		rawFunc := p.input[cp.pos:p.pos]
-		// Parse any expressions inside the function.
-		funcParser := NewParser()
-		pe, err := funcParser.Parse(rawFunc)
-		if err != nil {
-			return funcExpr{}, false, err
 		}
-		return funcExpr{raw: rawFunc, pe: pe}, true, nil
+		// Add any remaining unparsed string input to the parser.
+		p.add(nil)
```

Comment:
Just from our discussion earlier, instead of `p.add(nil)`, would the new `p.addBypass()` fit here? Otherwise, `addBypass` isn't used in this PR.

Also, just thinking about the naming, as `add` implicitly adds the bypass at the moment, it looks odd that (at a glance) the bypass is only explicitly added once. Perhaps this method should be renamed to something like `addRemainderAsBypass` or just `addRemainder`?

Reply:
I've added an `addRemainder` method to use for this case, its a good name for it.